### PR TITLE
Update Node e2e Core OS image to run systemd with CPU & Memory accounting enabled by default

### PR DESCRIPTION
**Pull request by vishh**
cc @derekwaynecarr 

For #26289

### CODE REVIEW

File: test/e2e_node/environment/setup_host.sh [hunk 1]
```diff
@@ -64,4 +64,13 @@ fi
 # delete init kubelet from containervm so that is doesn't startup
 if [ -f /etc/init.d/kubelet ]; then
   sudo rm /etc/init.d/kubelet
-fi
\ No newline at end of file
+fi
+
+# Setup systemd if required
+sudo mkdir -p /etc/systemd/system.conf.d
```
Comment (derekwaynecarr) [inline test/e2e_node/environment/setup_host.sh]: is there a reason you dropped the if check here: 
https://github.com/kubernetes/kubernetes/pull/26289/files#diff-6cfc24c58556b86ef8ff98e630f74f35R33

Comment (vishh) [inline test/e2e_node/environment/setup_host.sh]: The script is run manually and that's why I did not include any validation. My stack overflowed, and I missed your existing patch for updating this script. You can override this script with #26289.

**Comment by vishh** [comment]
@derekwaynecarr Yeah. I did not mean to include your patch in the PR. Removed your commit.

**Comment by derekwaynecarr** [comment]
@k8s-bot test this #ignore

**Comment by derekwaynecarr** [comment]
LGTM @pwittrock - any other comments?

**Comment by pwittrock** [comment]
Nope, LGTM

**Comment by vishh** [comment]
Rebased

**Pull request merged**